lightningd/notification.c

```diff
@@ -186,10 +187,16 @@ static void forward_event_notification_serialize(struct json_stream *stream,
 	struct forwarding *cur = tal(tmpctx, struct forwarding);
 	cur->channel_in = *in->key.channel->scid;
 	cur->msat_in = in->msat;
-	if (out) {
-		cur->channel_out = *out->key.channel->scid;
-		cur->msat_out = out->msat;
-		assert(amount_msat_sub(&cur->fee, in->msat, out->msat));
+	if (scid_out) {
+		cur->channel_out = *scid_out;
+		if (amount_out) {
+			cur->msat_out = *amount_out;
+			assert(amount_msat_sub(&cur->fee,
```

Comment:
In `NDEBUG` builds `assert` can remove the expression it has as argument, prefer instead `int result = foo(); assert(result);`.

Reply:
Yes, I didn't change this code, I only moved it.

I'll do a separate fix for post-0.8.1.